app/assets/src/components/Samples.jsx

```diff
@@ -225,12 +225,13 @@ class Samples extends React.Component {
     const hostList = this.state.selectedHostIndices.slice(0);
     let index;
     // check if the check box is checked or unchecked
+    let target_id = e.target.id.slice(5); // Remove 'host-' from front.
```

Comment:
How about you get target_id from a new attribute, which you could name data-host-id, via 
`let target_id = e.target.getAttribute("data-host-id")`?
It would be cleaner